File: Schedule/ClientApp/App/src/app/studying/models/FacultyDto.ts
import { Guid } from 'guid-typescript/dist/guid';
import { GroupDto } from './GroupDto';

export class FacultyDto {
  id: Guid;
  name: string;
  groups: GroupDto[];

  constructor(model: Partial<FacultyDto>) {
    this.id = model.id;
    this.name = model.name;
    this.groups = model.groups;
  }
}

File: Schedule.Core/DTO/Studying/FacultyDto.cs
using System;
using System.Collections.Generic;

namespace Schedule.Core.DTO.Studying
{
    public class FacultyDto
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public IEnumerable<GroupDto> Groups { get; set; }
    }
}
